src/createServer.js

```diff
@@ -1,7 +1,356 @@
+    }
+
+    await category.save();
+
+    res.json(category);
+  } catch (error) {
+    res
+      .status(400)
+      .json({ message: 'Error updating category', error: error.message });
+  }
+});
+
+router.delete('/categories/:id', async (req, res) => {
+  try {
+    const category = await Category.findByPk(req.params.id);
+
+    if (!category) {
+      return res.status(404).json({ message: 'Category not found' });
+    }
+
+    await category.destroy();
+
+    res.sendStatus(204);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: 'Error deleting category', error: error.message });
+  }
+});
+
+router.get('/expenses', async (req, res) => {
+  const { userId, category, from, to, categories } = req.query;
+
+  try {
+    const whereClause = {};
+
+    if (userId) {
+      whereClause.userId = userId;
+    }
+
+    if (category) {
+      whereClause.category = category;
+    }
+
+    if (categories) {
+      const categoryArray = Array.isArray(categories)
+        ? categories
+        : categories.split(',').map((cat) => cat.trim());
+
+      whereClause.category = categoryArray;
+    }
+
+    if (from && to) {
+      const start = new Date(from);
+      const end = new Date(to);
+
+      if (isNaN(start.getTime()) || isNaN(end.getTime())) {
+        return res.status(400).json({ message: 'Invalid date format' });
+      }
+
+      whereClause.spentAt = {
+        [Op.between]: [start, end],
+      };
+    }
+
+    const expenses = await Expense.findAll({ where: whereClause });
+
+    res.json(expenses);
+  } catch (error) {
+    res.status(500).json({ message: 'Error fetching expenses' });
+  }
+});
+
+router.get('/expenses/:id', async (req, res) => {
+  const id = Number(req.params.id);
+
+  try {
+    const expense = await Expense.findByPk(id);
+
+    if (!expense) {
+      return res.status(404).json({ message: 'Expense not found' });
+    }
+
+    res.json(expense);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: 'Error fetching expense', error: error.message });
+  }
+});
+
+router.post('/expenses', async (req, res) => {
+  const { title, amount, userId, category, note, spentAt } = req.body;
+
+  try {
+    if (!title || !amount || !userId) {
+      return res
+        .status(400)
+        .json({ message: 'Title, amount, and userId are required' });
+    }
+
+    const parsedAmount = parseFloat(amount);
+
+    if (isNaN(parsedAmount)) {
+      return res.status(400).json({ message: 'Invalid amount format' });
+    }
+
+    const user = await User.findByPk(userId);
+
+    if (!user) {
+      return res.status(404).json({ message: 'User not found' });
+    }
+
+    const newExpense = await Expense.create({
+      title,
+      amount: parsedAmount,
+      userId,
+      category: category || null,
+      note: note || null,
+      spentAt: spentAt || new Date().toISOString(),
+    });
+
+    return res.status(201).json(newExpense);
+  } catch (error) {
+    return res
+      .status(400)
```

Comment:
Consider using a 500 status code for server errors to maintain consistency with other error responses in the code.